src/D2L.Security.OAuth2/Keys/Default/D2LSecurityToken.cs

```diff
@@ -100,13 +97,32 @@ internal sealed partial class D2LSecurityToken : SecurityToken {
 			var key = GetKey();
 
 			if( key is RsaSecurityKey ) {
-				var csp = GetAsymmetricAlgorithm() as RSACryptoServiceProvider;
-				RSAParameters p = csp.ExportParameters( includePrivateParameters );
+				var rsaKey = key as RsaSecurityKey;
+
+				if( includePrivateParameters && !rsaKey.HasPrivateKey ) {
+					throw new Exception();
+				}
+
+				var parameters = rsaKey.Parameters;
+
+				if( !includePrivateParameters && rsaKey.HasPrivateKey ) {
+					var publicParameters = new RSAParameters();
+					publicParameters.Modulus = parameters.Modulus;
+					publicParameters.Exponent = parameters.Exponent;
+					parameters = publicParameters;
+				}
+
+				return new RsaJsonWebKey( KeyId, ValidTo, parameters );
+			} else if( key is ECDsaSecurityKey && !includePrivateParameters ) {
+				var ecDsaKey = key as ECDsaSecurityKey;
+				var ecDsaCng = ecDsaKey.ECDsa as ECDsaCng;
+				var cng = ecDsaCng?.Key;

```

Comment:
This area here is not cross-platform. It's as cross-platform as it was previously, but still not.

We might consider if we can have a `PrivateKeyProviders` output `JsonWebKey` or something, meaning it's already in place for us there, and then we can call `GetSignatureProvider` on that in order to get the `IDisposable` key things for usage.